specification/network/resource-manager/Microsoft.Network/stable/2024-03-01/networkWatcher.json

```diff
@@ -3192,6 +3192,10 @@
           "description": "ID of the storage account which is used to store the flow log.",
           "type": "string"
         },
+        "enabledFilteringCriteria": {
+          "description": "Optional field to filter network traffic logs based on SrcIP, SrcPort, DstIP, DstPort, Protocol, Encryption, Direction and Action. If not specified, all network traffic will be logged.",
```

Comment:
Can we convert this to Object Model? Say a Class Indicating first field as FilterField enum (SrcIP, SrcPort etc), Condition (say Equals, LessThan) and then Value (raw value)? Multiple expressions can also be combined in this style. That way clients can check which all filter fields are supported in SDK instead of relying on documentation?

Reply:
Thanks for the suggestion. As discussed over call, for now we will go with present approach. Since it is a major change in user experience, I'll take it up with my team to see if later we want to change it to Object model.